pkg/opendb/builtwith.go

```diff
@@ -0,0 +1,161 @@
+					LastDetected  int64    `json:"LastDetected"`
+					IsPremium     string   `json:"IsPremium"`
+					Categories    []string `json:"Categories,omitempty"`
+					Parent        string   `json:"Parent,omitempty"`
+				} `json:"Technologies"`
+				FirstIndexed int64  `json:"FirstIndexed"`
+				LastIndexed  int64  `json:"LastIndexed"`
+				Domain       string `json:"Domain"`
+				URL          string `json:"Url"`
+				SubDomain    string `json:"SubDomain"`
+			} `json:"Paths"`
+		} `json:"Result"`
+		Meta struct {
+			Majestic    int      `json:"Majestic"`
+			Umbrella    int      `json:"Umbrella"`
+			Vertical    string   `json:"Vertical"`
+			Social      []string `json:"Social"`
+			CompanyName string   `json:"CompanyName"`
+			Telephones  []string `json:"Telephones"`
+			Emails      []string    `json:"Emails"`
+			City        string   `json:"City"`
+			State       string   `json:"State"`
+			Postcode    string   `json:"Postcode"`
+			Country     string   `json:"Country"`
+			Names       []struct {
+				Name  string `json:"Name"`
+				Type  int    `json:"Type"`
+				Email string `json:"Email"`
+			} `json:"Names"`
+			ARank int `json:"ARank"`
+			QRank int `json:"QRank"`
+		} `json:"Meta"`
+		Attributes struct {
+			Employees    int `json:"Employees"`
+			MJRank       int `json:"MJRank"`
+			MJTLDRank    int `json:"MJTLDRank"`
+			RefSN        int `json:"RefSN"`
+			RefIP        int `json:"RefIP"`
+			Followers    int `json:"Followers"`
+			Sitemap      int `json:"Sitemap"`
+			GTMTags      int `json:"GTMTags"`
+			QubitTags    int `json:"QubitTags"`
+			TealiumTags  int `json:"TealiumTags"`
+			AdobeTags    int `json:"AdobeTags"`
+			CDimensions  int `json:"CDimensions"`
+			CGoals       int `json:"CGoals"`
+			CMetrics     int `json:"CMetrics"`
+			ProductCount int `json:"ProductCount"`
+		} `json:"Attributes"`
+		FirstIndexed int64  `json:"FirstIndexed"`
+		LastIndexed  int64  `json:"LastIndexed"`
+		Lookup       string `json:"Lookup"`
+		SalesRevenue int    `json:"SalesRevenue"`
+	} `json:"Results"`
+	Errors []struct {
+		Code	int    `json:"Code"`
+		Message string `json:"Message"`
+		Lookup  string `json:"Lookup"`
+	} `json:"Errors"`
+	Trust  string   `json:"Trust"`
+}
+
+// Builtwith retrieves from the url below some known subdomains.
+func BuiltWithSubdomains(target, apikey string, plain bool) []string {
+	result := []string{}
+
+	if !plain {
+		fmt.Println("Pulling data from BuiltWith")
+	}
+
+	client := http.Client{
+		Timeout: httpUtils.Seconds30,
+	}
+
+	fetchURL := fmt.Sprintf(
+		"https://api.builtwith.com/v21/api.json?&KEY=%s&LOOKUP=%s",
+		apikey, target,
+	)
+
+	resp, err := client.Get(fetchURL)
+	if err != nil {
+		return result
+	}
+
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return result
+	}
+
+	var bwrWrapper builtWithResultWrapper
+	if err := json.Unmarshal(body, &bwrWrapper); err != nil {
+		return result
+	}
+
+	for _, elem := range bwrWrapper.Results[0].Result.Paths {
```

Comment:
Got this error whie trying to use option `-bw`:

```
================ SCANNING SUBDOMAINS ================
Pulling data from BuiltWith
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/edoardottt/scilla/pkg/opendb.BuiltWithSubdomains({0x7ffdbf2f80f9, 0xa}, {0xc00058c030, 0x24}, 0xf9?)
	/home/edoardottt/test/scilla-ftr/pkg/opendb/builtwith.go:151 +0x4e8
github.com/edoardottt/scilla/pkg/runner.SubdomainSubcommandHandler({{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...)
	/home/edoardottt/test/scilla-ftr/pkg/runner/runner.go:376 +0xff2
github.com/edoardottt/scilla/pkg/runner.(*Runner).Execute(0xc00031fc70, 0x0?, 0x0?)
	/home/edoardottt/test/scilla-ftr/pkg/runner/runner.go:84 +0x1e8
main.main()
	/home/edoardottt/test/scilla-ftr/cmd/scilla/main.go:39 +0xc5
exit status 2
```

Reply:
I'm building the main package and running `./scilla subdomain -db -no-check -bw -target target.com` and getting a correct output.

What I think might be happening here is that for some domains entered the expected `Result` array in the response is coming up empty, so I think a check is in order here to just return an empty result and move on. Tested this case with a random domain `wqeqwd.com` got your error, and after adding in a fix the application flowed as normal.